core/src/main/java/io/crate/types/DataType.java

```diff
@@ -101,6 +104,14 @@ public void writeTo(StreamOutput out) throws IOException {
     }
 
     @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
shall this one and the next one be abstract? We probably want to enforce a user to implement them.